packages/govuk-frontend/src/govuk/components/accordion/accordion.mjs

```diff
@@ -14,7 +15,7 @@ import { I18n } from '../../i18n.mjs'
  * The state of each section is saved to the DOM via the `aria-expanded`
  * attribute, which also provides accessibility.
  */
-export class Accordion {
+export class Accordion extends Component {
```

Comment:
Oh, that's not something I had considered too much, just went with what felt right. `GOVUKComponent` feels very verbose, as well as misnamed (I'd think a `GOVUKComponent` would be part of the GOV.UK project). `GOVUKFrontendComponent` even more verbose. I'd be keen to keep `Component` for shortness.

For the `Error`, though, I think a little verbosity is useful to avoid confusion of what `throw new Error()` does (is it ours or a native error). If we shorten to `Error`, people would have to `import {Error as GOVUKFrontendError} from 'govuk-frontend'` to make their code clear, so we may as well take that step already 😊

Reply:
Feels strange having two approaches

How about `BaseError` and `BaseComponent`?

Reply:
I think it's important the error gets prefixed with `GOVUKFrontend`. It makes `instanceof` checks clearer: `instanceof GOVUKFrontendError` is more explicit than `instanceof BaseError`. This prefix would hopefully also make people think twice before extending it for their own purposes: we'd want them to throw their own kind of error if possible so there's a separation between service errors and our own library errors.

For the component bit, I think on the contrary it's important that its not labelled `GOVUKFrontend`. Down the road, we're imagining encouraging people to extend it. Creating `GOVUKFrontendComponent`s reads like creating "components of GOVUKFrontend", while it'd be just `Components` for their service.

As we haven't released yet, we can still tweak namings. Would you be fine moving naming discussion in its own issue while we implement the other errors? 😊

Reply:
Quick Slack decision?

Your point is great for `instanceof GOVUKFrontendComponent` too